Add basic user and session concepts, login form

Fnux opened the pull request.

Fnux committed:
Add login activity

Fnux committed:
Define basic data structures for users and sessions

Fnux committed:
Glue the login form to the new "session" backend

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/repository/UserRepository.java:
```diff
@@ -0,0 +1,55 @@
+        return USERS;
+    }
+
+    // FIXME: rewrite it to be compatible with older APIs revisions
```
What in this method isn't compatible with older APIs?

Fnux replied:
It looks like streams and `Optional.ofNullable/1` require API 24 (current minimum is 15, see #42). I didn't dig in the documentation for details, this PR is still WIP.

Fnux committed:
Add basic UI to account view, logout and navigation with home activity

Fnux committed:
Use res/strings.xml to store the titles of the login/account activities

Fnux committed:
Add padding to activities in eventSelector and userManager

Fnux committed:
Move user management backend to its own package

Fnux committed:
Fix invalid input error handling in login form

Fnux committed:
Add android tests to LoginActivity

Fnux committed:
Move LoginActivityTest to LoginActivity's package

Fnux committed:
Store login error messages into the project's string resources

See https://developer.android.com/guide/topics/resources/string-resource
for details.

Fnux committed:
Try to bump android API revision in Travis CI

Fnux committed:
Revert "Try to bump android API revision in Travis CI"

This reverts commit 89361cd5725152cad7d02ced7eb54ee1f6723a34.

Fnux committed:
Try to bump android API revision in Travis CI

Requesting the SDK in the components this time!

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/userManager/DisplayAccountActivity.java:
```diff
@@ -0,0 +1,44 @@
+
+        // We assume the user is logged in when the activity is opened.
+        TextView helpText = (TextView) findViewById(R.id.main_text);
+        helpText.setText("Logged as: " + Session.getCurrentUser().get().getEmail());
```
Session.getCurrentUser().get() might be null here (there are probably some cases where the user can glitch back to this activity after logging out)

Fnux replied:
We have to implement a generic permission system instead of a specific validation. This will have to be on another sprint.

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/userManager/LoginActivity.java:
```diff
@@ -0,0 +1,160 @@
+        setContentView(R.layout.activity_login);
+
+        mEmailView = (EditText)  findViewById(R.id.email_field);
+        mEmailView.setHint("Mail address");
```
Avoid hardcoded strings, put them in resources instead

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/userManager/LoginActivity.java:
```diff
@@ -0,0 +1,160 @@
+        mEmailView = (EditText)  findViewById(R.id.email_field);
+        mEmailView.setHint("Mail address");
+        mPasswordView = (EditText) findViewById(R.id.password_field);
+        mPasswordView.setHint("Password");
```
Avoid hardcoded strings, put them in resources instead

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/userManager/LoginActivity.java:
```diff
@@ -0,0 +1,160 @@
+        mPasswordView.setHint("Password");
+
+        mLoginButton = (Button) findViewById(R.id.login_button);
+        mLoginButton.setText("Sign in »");
```
Avoid hardcoded strings, put them in resources instead

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/eventSelector/EventPickingActivity.java:
```diff
@@ -52,5 +58,23 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
+        // Login button
+        Button loginButton = (Button) findViewById(R.id.login_button);
+        if (Session.isLoggedIn()) {
+            loginButton.setText("My account »");
```
Avoid hardcoded strings, put them in resources instead

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/eventSelector/EventPickingActivity.java:
```diff
@@ -52,5 +58,23 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
+        if (Session.isLoggedIn()) {
+            loginButton.setText("My account »");
+        } else {
+            loginButton.setText("Sign in »");
```
Avoid hardcoded strings, put them in resources instead

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/userManager/LoginActivity.java:
```diff
@@ -0,0 +1,160 @@
+    }
+
+    private boolean isEmailValid(String email) {
+        // FIXME: run a proper regex on the given email.
```
Don't think it's a good idea, as the perfect email validation regex doesn't exist.

zyuiop replied:
(actually not a proper review, I close)

Fnux replied:
The check was generated form an AndroidStudio template. I did not think of a perfect regex but more of the "general format" validation.

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/userManager/LoginActivity.java:
```diff
@@ -0,0 +1,160 @@
+        return email.contains("@");
+    }
+
+    private void showProgress(Boolean displayed) {
```
Take a `boolean` instead

Fnux replied:
Solved by d6ea85087bca781bb62f68fa1b14b1b87298664e.

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/userManager/LoginActivity.java:
```diff
@@ -0,0 +1,160 @@
+    }
+
+    private void showProgress(Boolean displayed) {
+       if (displayed) {
```
I'd rather do 
```java
mLoginButton.setEnabled(!displayed);
mProgressBar.setVisibility(enabled ? View.VISIBLE : View.INVISIBLE);
```

It's still very readable and avoids what looks like "duplicating code" (`setEnabled` and `setVisibility` appear twice). That's just an opinion though.

Fnux replied:
Solved by 2e9cd1a6a68ade17019b121fe3732e023029deb2.

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/userManager/LoginActivity.java:
```diff
@@ -0,0 +1,160 @@
+        @RequiresApi(api = Build.VERSION_CODES.N)
+        @Override
+        protected Boolean doInBackground(Void... params) {
+            // FIXME: we might not want to directly load the repository in this class
```
Yep, we probably want to `@Inject` it in the activity (no need for a view model here I guess)

zyuiop replied:
I'll find a solution for this

zyuiop replied:
I finally think it makes sense to have the repository directly in the Activity, as the task depends on the activity, and we might find some other uses for this repository directly in the activity in the future (actually, the task should probably be out of the activity and managed by the repository itself, but we will see once we will implement authentication against real async services)

zyuiop commented on app/src/main/java/ch/epfl/sweng/eventmanager/userManagement/InMemorySession.java:
```diff
@@ -0,0 +1,34 @@
+ */
+@Singleton
+public class InMemorySession {
+    private static InMemorySession instance = null;
```
It works for now, in the future we might want to use @Inject directly

zyuiop replied:
Not a proper review, I close

Fnux replied:
I'd like to keep things as simple as possible - and we don't need dependency injection here for now.

zyuiop reviewed (changes requested):
Regarding the "string litterals" I mentioned, you already defined most of them in strings.xml but still use the litterals directly, you probably just forgot to replace them.

Fnux committed:
LoginActivity: replace 'overkill' Boolean object by boolean primitive

Fnux committed:
Refactor LoginActivity's showProgress method

Fnux committed:
Move a few hardcoded strings to the strings resource file

Fnux committed:
Merge branch '42-drop-old-android-api-revisions' into 21-authentication

zyuiop committed:
Add dependency injection logic to the login activity

zyuiop committed:
Add an action on soft keyboard next/done buttons, and reduce methods length

zyuiop committed:
Merge branch 'master' into 21-authentication

Fnux commented on app/src/main/java/ch/epfl/sweng/eventmanager/ui/userManager/UserManagerModule.java:
```diff
@@ -0,0 +1,18 @@
+package ch.epfl.sweng.eventmanager.ui.userManager;
+
+import ch.epfl.sweng.eventmanager.ui.eventShowcase.MapActivity;
```
What is this `import` doing here?

zyuiop replied:
That's a fail

Fnux committed:
Travis CI: bump emulator SDK version to 28

zyuiop committed:
Remove useless import

zyuiop reviewed (dismissed):
Everything is ok for me!

zyuiop committed:
Merge branch 'master' into 21-authentication

zyuiop committed:
Travis: Downgrade SDK/emulator version to 26

zyuiop committed:
Try to fix travis testing issues

Found here: https://stackoverflow.com/questions/28751111/invalid-abi-armeabi-v7a-for-the-selected-target-with-google-apis

zyuiop committed:
Downgrade API LEVEL to 24 and add fancy variables in travis.yml

Fnux committed:
Restore Travis configuration using old emulator

Fnux committed:
Revert "Bump minimal supported android SDK version from 15 to 24"

This reverts commit 6117434382f43ef6d9c8bb5857875c4af272c653.

Fnux committed:
Remove optionals from the code (not supported by SDK revision < 24)

Fnux committed:
Fix Android manifest (and thus activity loading)

Fnux committed:
Remove streams from code (require android SDK >= 24)

Fnux committed:
Codeclimate: increase thresholds for method complexity and length

Fnux committed:
Merge branch '21-authentication'

Fnux committed:
Remove call to Object.equals (not supported by API 15)

zyuiop committed:
Fix activity_event_picking.xml layout issues

zyuiop committed:
Merge branch 'master' into 21-authentication
